spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/RuntimeCommands.java

```diff
@@ -0,0 +1,174 @@
+package org.springframework.cloud.dataflow.shell.command;
+
+import static org.springframework.shell.table.BorderSpecification.TOP;
+import static org.springframework.shell.table.BorderStyle.fancy_light;
+import static org.springframework.shell.table.BorderStyle.fancy_light_quadruple_dash;
+import static org.springframework.shell.table.CellMatchers.column;
+import static org.springframework.shell.table.CellMatchers.ofType;
+import static org.springframework.shell.table.CellMatchers.row;
+import static org.springframework.shell.table.SimpleHorizontalAligner.center;
+import static org.springframework.shell.table.SimpleVerticalAligner.middle;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.dataflow.rest.client.RuntimeOperations;
+import org.springframework.cloud.dataflow.rest.resource.ModuleInstanceStatusResource;
+import org.springframework.cloud.dataflow.rest.resource.ModuleStatusResource;
+import org.springframework.cloud.dataflow.shell.config.DataFlowShell;
+import org.springframework.shell.core.CommandMarker;
+import org.springframework.shell.core.annotation.CliAvailabilityIndicator;
+import org.springframework.shell.core.annotation.CliCommand;
+import org.springframework.shell.core.annotation.CliOption;
+import org.springframework.shell.table.BorderStyle;
+import org.springframework.shell.table.Table;
+import org.springframework.shell.table.TableBuilder;
+import org.springframework.shell.table.TableModel;
+import org.springframework.shell.table.TableModelBuilder;
+import org.springframework.shell.table.Tables;
+import org.springframework.stereotype.Component;
+
+/**
+ * Commands for displaying the runtime state of deployed modules.
+ *
+ * @author Eric Bottard
+ */
+@Component
+public class RuntimeCommands implements CommandMarker {
+
+	private static final String LIST_MODULES = "runtime modules";
+
+	@Autowired
+	private DataFlowShell dataFlowShell;
+
+	@CliAvailabilityIndicator({LIST_MODULES})
+	public boolean available() {
+		return dataFlowShell.getDataFlowOperations() != null;
+	}
+
+	@CliCommand(value = LIST_MODULES, help = "List runtime modules")
+	public Table list(
+			@CliOption(key = "summary", help = "whether to hide module instances details",
+					unspecifiedDefaultValue = "false", specifiedDefaultValue = "true") boolean summary,
+			@CliOption(key = {"moduleId", "moduleIds"}, help = "module id(s) to display, also supports '<group>.*' pattern") String[] moduleIds) {
+
+		Set<String> filter = null;
+		if (moduleIds != null) {
+			filter = new HashSet<>(Arrays.asList(moduleIds));
+		}
+
+		TableModelBuilder<Object> modelBuilder = new TableModelBuilder<>();
+		if (!summary) {
+			modelBuilder.addRow()
+					.addValue("Module Id / Instance Id")
+					.addValue("Unit Status")
+					.addValue("Nb of Instances / Attributes");
```

Comment:
Wow, I just learnt something (even though there is no 'o' in number :) )
Wow, I just (re)learned something again: both learnt and learned are correct.
Mind blown :boom: